packages/components/src/internal/components/domainproperties/dataclasses/DataClassDesigner.tsx

```diff
@@ -262,6 +328,83 @@ class DataClassDesignerImpl extends PureComponent<Props & InjectedBaseDomainDesi
         }
     };
 
+    updateAliasValue = (id: string, field: string, newValue: any): IParentAlias => {
+        const { model } = this.state;
+        const { parentAliases } = model;
+        return {
+            ...parentAliases.get(id),
+            isDupe: false, // Clear error because of change
+            [field]: newValue,
+        } as IParentAlias;
+    };
+
+    parentAliasChange = (id: string, field: string, newValue: any): void => {
+        const { model } = this.state;
+        const { parentAliases } = model;
+        const changedAlias = this.updateAliasValue(id, field, newValue);
+
+        const newAliases = parentAliases.set(id, changedAlias);
+        const newModel = {
+            ...model,
+            parentAliases: newAliases,
+        };
+        this.saveModel(newModel);
+    };
+
+    updateDupes = (id: string): void => {
+        const { model } = this.state;
+        if (!model) {
+            return;
+        }
+
+        const { parentAliases } = model;
+        const dupes = model.getDuplicateAlias();
```

Comment:
so much of this new code in this block is nearly a duplicate from `SampleTypeDesigner.tsx`. Any thoughts on if any of it can be shared or factored out as util functions?